Context::함수명 처럼 쓰일 수 있는 함수들에 static 적용

Web-Engine opened the pull request:
Context클래스의 함수 내부에서 $this 변수가 사용되지 않거나, is_a($this, 'Context') ? $self = $this : $self = self::getInstance(); 를 통해 $self 변수를 이용하는 경우 static을 붙임.

Web-Engine committed:
Context::함수명 처럼 쓰일 수 있는 함수들에 static 적용

Context클래스의 함수 내부에서 $this 변수가 사용되지 않거나, is_a($this, 'Context') ? $self = $this : $self = self::getInstance(); 를 통해 $self 변수를 이용하는 경우 static 변수를 붙임.

Web-Engine committed:
$this 변수를 $self로 변경

is_a($this, 'Context') ? $self = $this : $self = self::getInstance();을 통해 $self 변수를 생성하엿으나, $self를 사용하지 않고 $this가 사용됨.
=> 이 함수를 Context::함수명 처럼 사용시 오류의 소지가 있음

kijin commented:
우와! 드디어 PHP 5 느낌이 좀 나네요 ㅎㅎ

하시는 김에 정식으로 public까지 다 붙여주는 건 어떨까요? (기존에 public 속성을 가진 메소드가 딱 하나 있었는데, static public으로 순서가 잘못 들어갔네요. public static이 정석입니다.)

참고: [PSR-2](http://www.php-fig.org/psr/psr-2/)

> Visibility MUST be declared on all properties and methods; abstract and final MUST be declared before the visibility; static MUST be declared after the visibility.

Web-Engine commented:
@kijin 좋은 내용 감사드립니다.
public도 붙여서 pull request 넣도록 하겠습니다. ^^

Web-Engine committed:
함수 앞에 public 추가

함수 앞에 public을 추가 및 static public으로 되어 있던 부분을 public static으로 변경.
그리고, close 함수도 static으로 변경

misol commented:
이 PR을 Apache, PHP/5.5.15 기반, 그리고 nginx, PHP/5.5.9 기반에서 테스트 했을 때 서버 스크립트 문제가 발생하지 않았습니다.

misol commented:
이건 언제쯤 반영될지..
